src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs

```diff
@@ -23,6 +23,289 @@ public class RefLocalsAndReturnsTests : CompilingTestBase
+            var comp = CreateCompilationRef(@"
+class C
+{
+    void M()
+    {
+        void L(ref int p) { }
+
+        L(ref 42);
+        int x = 0;
+        ref readonly int xr = ref x;
+        L(xr);
+        L(ref xr);
+
+        ref readonly int L2() => ref (new int[1])[0];
+
+        L(L2());
+        L(ref L2());
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (8,15): error CS1510: A ref or out value must be an assignable variable
+                //         L(ref 42);
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, "42").WithLocation(8, 15),
+                // (11,11): error CS1620: Argument 1 must be passed with the 'ref' keyword
+                //         L(xr);
+                Diagnostic(ErrorCode.ERR_BadArgRef, "xr").WithArguments("1", "ref").WithLocation(11, 11),
+                // (12,15): error CS1510: A ref or out value must be an assignable variable
+                //         L(ref xr);
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, "xr").WithLocation(12, 15),
+                // (16,11): error CS1620: Argument 1 must be passed with the 'ref' keyword
+                //         L(L2());
+                Diagnostic(ErrorCode.ERR_BadArgRef, "L2()").WithArguments("1", "ref").WithLocation(16, 11),
+                // (17,15): error CS8406: Cannot use method 'L2()' as a ref or out value because it is a readonly variable
+                //         L(ref L2());
+                Diagnostic(ErrorCode.ERR_RefReadonlyNotField, "L2()").WithArguments("method", "L2()").WithLocation(17, 15));
+        }
+
+        [Fact]
+        public void AssignRefReadonlyLocalToRefLocal()
+        {
+            var comp = CreateCompilationRef(@"
+class C
+{
+    void M()
+    {
+        ref readonly int L() => ref (new int[1])[0];
+
+        ref int w = ref L();
+        ref readonly int x = ref L();
+        ref int y = x;
+        ref int z = ref x;
+    }
+}");
+            comp.VerifyDiagnostics(
+                // (8,25): error CS8406: Cannot use method 'L()' as a ref or out value because it is a readonly variable
+                //         ref int w = ref L();
+                Diagnostic(ErrorCode.ERR_RefReadonlyNotField, "L()").WithArguments("method", "L()").WithLocation(8, 25),
+                // (10,17): error CS8172: Cannot initialize a by-reference variable with a value
+                //         ref int y = x;
+                Diagnostic(ErrorCode.ERR_InitializeByReferenceVariableWithValue, "y = x").WithLocation(10, 17),
+                // (10,21): error CS1510: A ref or out value must be an assignable variable
+                //         ref int y = x;
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, "x").WithLocation(10, 21),
+                // (11,25): error CS1510: A ref or out value must be an assignable variable
+                //         ref int z = ref x;
+                Diagnostic(ErrorCode.ERR_RefLvalueExpected, "x").WithLocation(11, 25)
+            );
```

Comment:
Need a test that essentially does the following

ref readonly int x = ref 1;

This is legal when done as an argument to a ref readonly parameter. Need to be clear if this is okay / not okay when done in this context.

Reply:
This is prohibited for now. I'll add test. It does seem a little annoying that this is prohibited for locals but allowed for calls. We may want to relax that restriction and auto-spill like we do for arguments, but I'm not convinced that needs to be blocking for this PR.